test/python/v1/rebuild/test_bdd_rebuild.py

```diff
@@ -0,0 +1,268 @@
+import pytest
+from pytest_bdd import given, scenario, then, when, parsers
+
+import os
+import subprocess
+import time
+
+from v1.mayastor import container_mod, mayastor_mod
+from v1.volume import Volume
+
+import grpc
+import nexus_pb2 as pb
+
+
+def megabytes(n):
+    return n * 1024 * 1024
+
+
+def find_child(nexus, uri):
+    for child in nexus.children:
+        if child.uri == uri:
+            return child
+    return None
+
+
+def convert_nexus_state(state):
+    STATES = {
+        "UNKNOWN": pb.NexusState.NEXUS_UNKNOWN,
+        "ONLINE": pb.NexusState.NEXUS_ONLINE,
+        "DEGRADED": pb.NexusState.NEXUS_DEGRADED,
+        "FAULTED": pb.NexusState.NEXUS_FAULTED,
+    }
+    return STATES[state]
+
+
+def convert_child_state(state):
+    STATES = {
+        "UNKNOWN": pb.ChildState.CHILD_UNKNOWN,
+        "ONLINE": pb.ChildState.CHILD_ONLINE,
+        "DEGRADED": pb.ChildState.CHILD_DEGRADED,
+        "FAULTED": pb.ChildState.CHILD_FAULTED,
+    }
+    return STATES[state]
+
+
+def convert_child_action(state):
+    ACTIONS = {
+        "OFFLINE": pb.ChildAction.Offline,
+        "ONLINE": pb.ChildAction.Online,
+    }
+    return ACTIONS[state]
+
+
+@scenario("features/rebuild.feature", "running rebuild")
+def test_running_rebuild():
+    "Running rebuild."
+
+
+@scenario("features/rebuild.feature", "stopping rebuild")
+def test_stopping_rebuild():
+    "Stopping rebuild."
+
+
+@scenario("features/rebuild.feature", "pausing rebuild")
+def test_pausing_rebuild():
+    "Pausing rebuild."
+
+
+@scenario("features/rebuild.feature", "resuming rebuild")
+def test_resuming_rebuild():
+    "Resuming rebuild."
+
+
+@scenario("features/rebuild.feature", "setting a child ONLINE")
+def test_setting_a_child_online():
+    "Setting a child ONLINE."
+
+
+@scenario("features/rebuild.feature", "setting a child OFFLINE")
+def test_setting_a_child_offline():
+    "Setting a child OFFLINE."
+
+
+@pytest.fixture(scope="module")
+def local_files():
+    files = [f"/tmp/disk-rebuild-{base}.img" for base in ["source", "target"]]
+    for path in files:
+        subprocess.run(
+            ["sudo", "sh", "-c", f"rm -f '{path}' && truncate -s 64M '{path}'"],
+            check=True,
+        )
+    yield
+    for path in files:
+        subprocess.run(["sudo", "rm", "-f", path], check=True)
+
+
+@pytest.fixture(scope="module")
+def source_uri(local_files):
+    yield "aio:///tmp/disk-rebuild-source.img?blk_size=4096"
+
+
+@pytest.fixture(scope="module")
+def target_uri(local_files):
+    yield "aio:///tmp/disk-rebuild-target.img?blk_size=4096"
+
+
+@pytest.fixture(scope="module")
+def nexus_uuid():
+    yield "2c58c9f0-da89-4cb9-8097-dc67fa132493"
+
+
+@pytest.fixture(scope="module")
+def nexus_name():
+    yield "nexus-1"
+
+
+@pytest.fixture(scope="module")
+def mayastor_instance(mayastor_mod):
+    yield mayastor_mod["ms0"]
+
+
+@pytest.fixture(scope="module")
+def find_nexus(mayastor_instance):
```

Comment:
word of caution: if the rebuild is "too fast", it may complete before you list the nexus, and thus the test cases may fail as the states are now different.. Instead I recommend caching the nexus returned by the when clauses.
(Note this is something that has happened to us and caused intermittent failures in the CI)